drivers/console/semihosting_console.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2018 Josef Gajdusek <atx@atx.name>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <kernel.h>
+#include <misc/printk.h>
+#include <device.h>
+#include <init.h>
+#include <string.h>
+
+extern void __printk_hook_install(int (*fn)(int));
+extern void __stdout_hook_install(int (*fn)(int));
+
+/* See the Semihosting for AArch32 and AArch64 document for reference */
+#define SYS_OPEN 0x01
+#define SYS_WRITE 0x05
+
+#define OPEN_MODE_R		0
+#define OPEN_MODE_RB	1
+#define OPEN_MODE_RP	2
+#define OPEN_MODE_RPB	3
+#define OPEN_MODE_W		4
+#define OPEN_MODE_WB	5
+#define OPEN_MODE_WP	6
+#define OPEN_MODE_WPB	7
+#define OPEN_MODE_A		8
+#define OPEN_MODE_AB	9
+#define OPEN_MODE_AP	10
+#define OPEN_MODE_APB	11
+
+static char buffer[CONFIG_SEMIHOSTING_BUFFER_SIZE];
+static u32_t buffer_pos;
+static int stderr_handle = -1;
+
+static void semihosting_flush_buffer(struct k_timer *timer);
+static K_TIMER_DEFINE(send_timer, semihosting_flush_buffer, NULL);
+
+
+static int semihosting_request(u32_t command, void *message)
```

Comment:
I wonder if we should move this to a header file in order to keep the code compatible with multiple compilers (i.e. LLVM down the line).  Maybe in `include/arch/`?